hca/util/__init__.py

```diff
@@ -447,14 +446,24 @@ def _get_jwt_from_service_account_credentials(self):
                                 algorithm='RS256').decode()
         return signed_jwt, exp
 
+    def expired_token(self):
```

Comment:
When the token doesn't exist this method returns `False`, indicating a non-expired token. This may be confusing behavior. It may be preferable to raise in this scenario.

Reply:
If `self._authenticated_session` doesn't exist, that's fine and it doesn't have an expired token, it has no session (and will create a session outside of this function).  If `self._authenticated_session.token` doesn't exist or is `None`, it will error.

I can enforce that `expires_at` always exists in the token, which it always should and seems like a sensible check.

Reply:
I believe `self._authenticated_session.token` is initialized to `dict()`, in which case this method will confusingly return `False` if it is used outside of the context of  _your_ use case, which a future coder may plausibly do. Expecting `expires_at` to exist in the token would fix this issue.

Reply:
Changes made.

Reply:
It inits to `None` instead of `dict()`: https://github.com/HumanCellAtlas/dcp-cli/blob/3789edffb6804f02290a905dab093dde6cd7884b/hca/util/__init__.py#L233

Reply:
Related to Brian's comment - I find it useful to write access control/security functions such that the default boolean returned is "False"/the default outcome is to deny access, and an explicit conditional check must be met for the function to return True and allow access. For example, this function could be renamed to `token_is_active()` or similar, and would return False by default (equivalent to assuming every token is expired) and True if the required conditions are met.

Reply:
The security check here is more of a courtesy. Expired tokens cannot be 100% enforced on the client side. I recommend going for readability.